src/components/inputs/text/ArweavetransactionIdInput/ArweavetransactionIdInput.tsx

```diff
@@ -0,0 +1,131 @@
+import { useEffect } from 'react';
+
+import { ArweaveTransactionID } from '../../../../types';
+import {
+  ARNS_TX_ID_ENTRY_REGEX,
+  ARNS_TX_ID_REGEX,
+} from '../../../../utils/constants';
+
+const validityObject = {
+  'Is a transaction ID': {
+    result: undefined, // success, error, warning, loading (validating), undefined (grey icon)
+    error: '', // error messages from validation function - tooltip to view error
+  },
+  'Has approved contract source code': {
+    result: undefined,
+    error: '',
+  },
+  'Has 50 confirmations': {
+    result: undefined,
+    error: '',
+  },
+  'Is owned by user': {
+    result: undefined,
+    error: '',
+  },
+};
+
+function ArweaveTransactionIdInput({
+  className,
+  disabled,
+  value,
+  setValue,
+  setValidation,
+  setIsValid,
+  customStyle,
+  placeholder,
+}: {
+  className?: string;
+  disabled?: boolean;
+  value: string;
+  setValue: (text: string) => void;
+  setValidation: (validation: any) => void;
+  setIsValid: (validity: boolean) => void;
+  customStyle?: any;
+  placeholder?: string;
+}) {
+  useEffect(() => {
+    setValidation([
+      // set default validations on load
+      {
+        'Is a transaction ID': {
+          result: undefined,
+          error: '',
+        },
```

Comment:
i'm not following the need for this object it. we could simplify by having just a `Set` (or array) of validation errors